airflow/models/xcom.py

```diff
@@ -221,28 +267,51 @@ def delete(cls, xcoms, session=None):
     @provide_session
     def clear(
         cls,
-        execution_date: pendulum.DateTime,
-        dag_id: str,
-        task_id: str,
+        execution_date: Optional[pendulum.DateTime] = None,
+        dag_id: str = None,
+        task_id: str = None,
+        run_id: str = None,
         session: Session = None,
     ) -> None:
         """
         Clears all XCom data from the database for the task instance
 
+        ``run_id`` and ``execution_date`` are mutually exclusive.
+
         :param execution_date: Execution date for the task
-        :type execution_date: pendulum.datetime
+        :type execution_date: pendulum.datetime or None
         :param dag_id: ID of DAG to clear the XCom for.
         :type dag_id: str
         :param task_id: Only XComs from task with matching id will be cleared.
         :type task_id: str
+        :param run_id: Dag run id for the task
+        :type run_id: str or None
         :param session: database session
         :type session: sqlalchemy.orm.session.Session
         """
-        session.query(cls).filter(
+        # Given the historic order of this function (execution_date was first argument) to add a new optional
+        # param we need to add default values for everything :(
+        if not dag_id:
+            raise TypeError("clear() missing required argument: dag_id")
```

Comment:
Should this and L297 be `ValueError` too?

Reply:
I think these should all be TypeError.